geonode/layers/download_handler.py

```diff
@@ -51,9 +51,39 @@ def get_download_response(self):
         that allow the resource download
         """
         resource = self.get_resource()
+        if not resource:
+            raise Http404("Resource requested is not available")
         response = self.process_dowload(resource)
         return response
 
+    @property
+    def is_default(self):
+        return "DatasetDownloadHandler" in str(self)
```

Comment:
Probably the logic to select the "default" handler should be outside the handler itself.
The serializer should select the default handler. Something lilke:

- the serializer looks for a `settings.DEFAULT_DATASET_DOWNLOAD_HANDLER` with the path of the handler module
- if it's set, ok
- otherwise, it will select the first item from the `settings.DATASET_DOWNLOAD_HANDLERS` list